src/test/java/org/assertj/core/api/AutoCloseableSoftAssertionsTest.java

```diff
@@ -95,7 +95,7 @@ public void should_be_able_to_catch_exceptions_thrown_by_all_proxied_methods() {
       softly.assertThat(new int[] { 24 }).isEqualTo(new int[] { 25 });
 
       softly.assertThat((Iterable<String>) list("26")).isEqualTo(list("27"));
-      softly.assertThat(list("28").iterator()).contains("29");
+      softly.assertThat(list("28")).contains("29");
```

Comment:
Not sure to get this change, we still want an assertion involving `Iterator`

Reply:
I am not sure if I understand you correctly. Should I add a consuming `contains` assertion or not?

Reply:
sorry I was not clear!
what I we need here is to add an `Iterator` assertion (`hasNext` will do) to make sure they work with soft assertions.
It probably easier to add it after `softly.assertThat("48").isEqualTo("49");` to avoid changing the error messages order in the catch block.